app/bundles/CoreBundle/Doctrine/Mapping/ClassMetadataBuilder.php

```diff
@@ -465,4 +461,24 @@ public function isIndexedVarchar(string $name, string $type): bool
     {
         return Types::STRING === $type || isset($this->getClassMetadata()->table['indexes'][$name]);
     }
+
+    /**
+     * Adds Index with options.
+     *
+     * @param list<string>         $columns
+     * @param array<string, mixed> $options
+     *
+     * @return $this
+     */
+    public function addIndexWithOptions(array $columns, string $name, array $options)
```

Comment:
On the other hand the new methods should have param and return types :)
```suggestion
     */
    public function addIndexWithOptions(array $columns, string $name, array $options): ClassMetadataBuilder
```

Reply:
hah, how does the pipeline not cover this? both locally as the github pipelines I have no issues in phpstan or php-cs-fixer

Reply:
We'll have to add the rule and rebuild the PHPSTAN baseline. It's on my todo list for some time, but wouldn't mind if you get to it first :)